src/spec.cr

```diff
@@ -23,6 +23,25 @@ module Shards
       end
     end
 
+    class Library
+      property soname : String
```

Comment:
I'm on the fence, `soname` future proofs a bit in case we ever want to allow `pkg-config` names too or the like. But if you have a strong preference for `name` I'm ok with changing it.